### ISSUE TITLE
Limit the need for repeatedly re-entering the OAuth credentials when using amplify init

### ISSUE THREAD

**Issue by PeteDuncanson** [aws-amplify/amplify-cli#6430]
Often when using the CLI I'll be asked to provide the OAuth url and secret for my App. This step is always a blocking step, you can't skip it and come back to it later.

Through the whole life of our app we've used the same details but digging them out everytime hasn't been easy and has now lead to us storing at hand for quick retrieval.

Is there a reason that these can't be stored somewhere per environment rather than repeatedly asked for? The values appear to be stored anyway locally so is there not a way to keep them rather than keep deleting them?

**Comment by PeteDuncanson** [comment]
This is what I mean:
![image](https://user-images.githubusercontent.com/1674590/105088983-73d60e80-5a94-11eb-9b96-3980f2d0a271.png)

**Comment by mattlanham** [comment]
Also experiencing this issue, can't they be stored in AWS secrets manager or similar and referenced from there on a per app basis?

**Comment by PeteDuncanson** [comment]
As mentioned by @avlonder in #6481 it seems ``amplify pull`` can ask for repeated credentials too, I could understand why as you might have different for different environments but could it not keep those separately keyed by environment. So if you switch to "dev" then it looks to see if it has any "dev" credentials, if so it uses them and if not then it can ask you for them. Think that would be a good way around this. Currently its a debugging/development flow killer as is.

**Comment by ammarkarachi** [comment]
@PeteDuncanson this looks like an issue on the CLI will mark this as a bug

**Comment by ammarkarachi** [comment]
@PeteDuncanson This should be available in the latest release

**Comment by PeteDuncanson** [comment]
@ammarkarachi going to have to reopen this one I think, just hit a snag with the fix I think.

Totally new laptop and a full fresh checkout of my project. Did an `amplify init` and didn't get asked for the OAuth stuff which was promising. However when doing a push I got the following error:

UPDATE_FAILED      HostedUIProvidersCustomResourceInputs Custom::LambdaCallout Wed Apr 14 2021 14:27:12 GMT+0100 (British Summer Time) Failed to update resource. See the details in CloudWatch Log Stream: 2021/04/14/[$LATEST]xxxxxxxx

This kills the push and forces a rollback. So I can't deploy. Some digging around and I found your two fixes #6485 #6739 (thanks for those) which mention the HostedUI stuff so I'm thinking they are related.

As a work around I had to run `amplify update auth` and reenter my OAuth credentials again and then do another push which then works and unblocks me...took me 2 hours to work that out though, hope it helps someone in the meantime.

I'm not sure if these details are stored in the cloud somewhere or in a local file that isn't getting updated or something?
